ratelimits/bucket.go

```diff
@@ -320,19 +313,29 @@ func (builder *TransactionBuilder) CertificatesPerDomainTransactions(regId int64
 			}
 			txns = append(txns, txn)
 
-			// Add a spend-only transaction for each per domain bucket.
-			txn, err = newSpendOnlyTransaction(perDomainLimit, perDomainBucketKey, 1)
-			if err != nil {
+			perDomainLimit, err := builder.getLimit(CertificatesPerDomain, perDomainBucketKey)
+			if err != nil && !errors.Is(err, errLimitDisabled) {
 				return nil, err
+			} else {
+				// Add a spend-only transaction for each per domain bucket.
+				txn, err = newSpendOnlyTransaction(perDomainLimit, perDomainBucketKey, 1)
```

Comment:
Reading the conditional in reverse (and applying DeMorgan's Laws), we hit the `else` clause when `err == nil || errors.Is(err, errLimitDisabled)`.

I think from previous comments in the review the intent was to not add any transaction at all when the limit is disabled. E.g.:

```
if errors.Is(err, errLimitDisabled) {
  continue
}
if err != nil {
  return nil, err
}
// Add a spend-only transaction ...
txn, err = ...
```